src/app.rs

```diff
@@ -1711,6 +1717,19 @@ impl ProfApp {
         result
     }
 
+    fn pan(cx: &mut Context, percent: PercentageInteger, dir: PanDirection) {
+        if percent.value() == 0 {
+            return;
+        }
+
+        let duration: i64 = percent.apply_to(cx.view_interval.duration_ns());
+        let sign: i64 = match dir {
+            PanDirection::Left => -1,
+            PanDirection::Right => 1,
+        };
+        cx.view_interval = cx.view_interval.translate(duration * sign)
```

Comment:
If you're going to do this without going through `zoom()`, you need to copy these lines:

```rust
        cx.interval_state.start_buffer = cx.view_interval.start.to_string();
        cx.interval_state.stop_buffer = cx.view_interval.stop.to_string();
        cx.interval_state.start_error = None;
        cx.interval_state.stop_error = None;
```

Maybe we should abstract that out into a different function so that we don't duplicate this behavior.

Reply:
https://github.com/StanfordLegion/prof-viewer/pull/27/commits/a0d572e7d68c75f6aab9a3a522bf521c909b7cb4